File: adhoc/2061/2061.cpp
#include<iostream>
#include<string>
#include<functional>
#include<tuple>
#include <sstream>

using namespace std;

enum class Acoes {
    fechou,
    clicou,
    unknown
};

Acoes hashString(const std::string& str){
    if(str == "fechou") return Acoes::fechou;
    if(str == "clicou") return Acoes::clicou;
    return Acoes::unknown;
}

std::tuple<int, int> input_dados(){
    string input;
    char delimiter = ' ';
    string t;
    int count = 0; int first = 0; int second = 0;

    getline(cin, input);
    
    stringstream ss(input);

    while(std::getline(ss, t, delimiter)){
        if (count == 0){
            first = std::stoi(t);
        } else if (count == 1){
            second = std::stoi(t);
            break;
        }
        count++;
    }

    return std::make_tuple(first, second);
}

class Browser{
    public:
        Browser(int abas) : _abas(abas) {}

        void acao(const std::string& acoes){
            switch (hashString(acoes)){
                case Acoes::fechou:
                    _abas += 1;                
                    break;
                
                case Acoes::clicou:
                    _abas -=1;
                default:
                    break;
            }
        }

        int getAbas() const {
            return _abas;
        }

    private:
        int _abas;
};

int main(){
    auto [abas_inicial, sequencia] = input_dados();

    Browser browser(abas_inicial);

    for (int i = 0; i < sequencia; i++){
        string acao;
        cin >> acao;
        browser.acao(acao);
    }

    cout << browser.getAbas() << endl;

    return 0;
}

File: adhoc/2061/2061.py
'''
clicar em propaganda -> apaga aba sem abrir outra
fecha aba -> abre duas abas

inputs numero inicial sequencia de ações
fechou ou clicou

pergunta: quantas abas sobraram
'''
    
class Browser:
    def __init__(self, abas:int):
        self.abas = abas
    
    def acao(self, acoes:str):
        match acoes:
            case "clicou":
                self.abas -= 1
            case "fechou":
                self.abas += 1

def input_dados():
    string_input = input()
    abas_inicial, sequencia = string_input.split(' ')
    abas_inicial = int(abas_inicial)
    sequencia = int(sequencia)
    
    if abas_inicial < 0:
        print("ERRO: Abas_inicial menor que zero.")
        return 0, 0
    if sequencia > 500:
        print("ERRO: Muitas ações.")
        return 0, 0
    if sequencia < 0:
        print("ERRO: Ações negativas.")
        return 0, 0
    return abas_inicial, sequencia

def main():
    abas_inicial, sequencia = input_dados()
    
    browser = Browser(abas= abas_inicial)
    
    while sequencia !=0:
        acoes = str(input())
        browser.acao(acoes = acoes)
        sequencia -= 1
        
    return print(browser.abas)

if __name__ == "__main__":
    main()
